example_client/multi_inputs.py

```diff
@@ -0,0 +1,280 @@
+def parse_output(thr_id, res, frame):
+  for batch, data in enumerate(res):
+    pred = data[0]
+    for values in enumerate(pred):
+      # tuple
+      index = values[0]
+      l_pred = values[1]
+
+      # actual predictions
+      img_id = l_pred[0]
+      label = l_pred[1]
+      conf = l_pred[2]
+      x_min = l_pred[3]
+      y_min = l_pred[4]
+      x_max = l_pred[5]
+      y_max = l_pred[6]
+
+      # preventing any wrong array indexing (for RMNet)
+      if label > 4:
+        log.error('Unsupported class detected in camera {}'.format(thr_id))
+        label = 4
+
+      # Do you want confidence level to be passed from command line?
+      if img_id != -1 and conf >= CONFIDENCE_THRESHOLD:
+        # draw the bounding boxes on the frame
+        height, width = frame.shape[:2]
+        cv2.rectangle(frame, (int(width * x_min), int(height * y_min)),
+              (int(width * x_max), int(height * y_max)), COLORS[int(label)], 2)
+        cv2.putText(frame, str(CLASSES[int(label)]), (int(width * x_min)-10,
+              int(height * y_min)-10), cv2.FONT_HERSHEY_SIMPLEX, 0.5,
+              COLORS[int(label)], 2)
+  return frame
+
+# This is common for both the camera & video files
+def thread_function(thr_id, network_name, input_layer, output_layer, input_dimension,
+                    stub, thr_lock, disp_buf, src_type, src_name):
+  if src_type == "Camera":
+    # UVC camera init - camera threads always come first and we use it
+    # to generate the camera indexes
+    cam = cv2.VideoCapture(thr_id)
+    if not (cam.isOpened()):
+      log.error("Failed to open the UVC camera {}".format(thr_id))
+      return
+
+    cam.set(cv2.CAP_PROP_FRAME_WIDTH, CAM_WIDTH)
+    cam.set(cv2.CAP_PROP_FRAME_HEIGHT, CAM_HEIGHT)
+    # not all UVC cameras honor below request
+    cam.set(cv2.CAP_PROP_FPS, CAM_FPS)
+    # If your camera sends other than MJPEG, change below
+    cam.set(cv2.CAP_PROP_FOURCC, cv2.VideoWriter_fourcc(*"MJPG"))
+  elif src_type == "Video":
+    # Assumption: src_name will be valid
+    cam = cv2.VideoCapture(src_name)
+
+  # stats
+  fps = 0                   # camera fps
+  inf_fps = 0               # inference fps
+  dropped_fps = 0           # dropped frame fps
+  cam_start_time = time()
+
+  request = predict_pb2.PredictRequest()
+  # Note: Pls maintain the same name while launching ovms docker container
+  request.model_spec.name = network_name
+
+  global exit_ok
+  while exit_ok == False:
+    ret, frame = cam.read()
+
+    if src_type == "Video":
+      # restart the video file when it reaches the end
+      if not ret:
+        cam.set(cv2.CAP_PROP_POS_FRAMES, 0)
+        continue
+      # normalize the video frame dimension to that of the camera
+      else:
+        # to maintain the frame inferencing parity with the cameras, lets sleep
+        # here to maintain cam_fps speed
+        sleep((1000 / CAM_FPS) / 1000)
+        frame = cv2.resize(frame, (CAM_WIDTH, CAM_HEIGHT))
```

Comment:
I know I mentioned it before, but I'm still not sure if this statement will be effective as you have line 165:
`image = cv2.resize(frame, (input_dimension, input_dimension))`
So, here you resize your frame to (CAM_WIDTH, CAM_HEIGHT) and then again to (input_dimension, input_dimension).
Given that you don't use `frame` between those lines, it seems to me that this first resize is not really effective.

Reply:
There are two things that are happening here:
a) Network needs input in certain resolution - line 165 which you refer above
b) The results are displayed on windows using the inference results - In this case, camera is already asked to provide the input "frame" in CAM_WIDTH & CAM_HEIGHT dimension. However our example also supports the video files. To maintain the output window parity across the camera and video files, we do an extra resize when we read one frame from the video file. This is "frame" which will be used to show the result.

This is the reason why two resizes exist. It does have effect. If you comment out line 153 above, the output frame from the video file would be displayed with whatever dimension it has been read. I will remove it for your satisfaction although the output window looks inconsistent(esp on small monitors)